packages/bs-neox/src/BSNeoX.ts

```diff
@@ -28,22 +42,155 @@ export class BSNeoX<N extends string = string> extends BSEthereum<N, TBSNeoXNetw
   }
 
   setNetwork(network: TBSNetwork<TBSNeoXNetworkId>) {
-    const availableURLs = BSNeoXConstants.RPC_LIST_BY_NETWORK_ID[network.id] || []
+    const rpcNetworkUrls = BSNeoXConstants.RPC_LIST_BY_NETWORK_ID[network.id] || []
+    const isValidNetwork = BSUtilsHelper.validateNetwork(network, this.availableNetworks, rpcNetworkUrls)
 
-    const isValidNetwork = BSUtilsHelper.validateNetwork(network, this.availableNetworks, availableURLs)
     if (!isValidNetwork) {
       throw new Error(`Network with id ${network.id} is not available for ${this.name}`)
     }
 
     this.network = network
-    this.availableNetworkURLs = availableURLs
+    this.rpcNetworkUrls = rpcNetworkUrls
 
     this.nftDataService = new GhostMarketNDSNeoX(this)
     this.explorerService = new BlockscoutESNeoX(this)
     this.exchangeDataService = new FlamingoForthewinEDSNeoX(this)
     this.neo3NeoXBridgeService = new Neo3NeoXBridgeService(this)
     this.blockchainDataService = new BlockscoutBDSNeoX(this)
     this.tokenService = new TokenServiceEthereum()
-    this.walletConnectService = new WalletConnectServiceEthereum(this)
+    this.walletConnectService = new WalletConnectServiceNeoX(this)
+  }
+
+  async transfer(params: TTransferParam<N>): Promise<string[]> {
+    const isAntiMevRpcNetworkUrl = BSNeoXConstants.ANTI_MEV_RPC_LIST_BY_NETWORK_ID[this.network.id].some(
+      rpcNetworkUrl => rpcNetworkUrl === this.network.url
+    )
+
+    if (!isAntiMevRpcNetworkUrl) return await super.transfer(params)
+
+    const signer = await this.generateSigner(params.senderAccount)
+    let nonce = await signer.getTransactionCount()
+
+    if (isNaN(nonce)) {
+      throw new Error('Invalid nonce')
+    }
+
+    const chainId = +this.network.id
+
+    if (isNaN(chainId)) {
+      throw new Error('Invalid chainId')
+    }
+
+    const transactionHashes: string[] = []
+    let error: Error | undefined
+
+    for (const intent of params.intents) {
+      try {
+        const { transactionParams, gasPrice } = await this._buildTransferParams(intent)
+        let gasLimit: ethers.BigNumberish
+
+        try {
+          gasLimit = await signer.estimateGas(transactionParams)
+        } catch {
+          gasLimit = BSEthereumConstants.DEFAULT_GAS_LIMIT
+        }
+
+        await BSPromisesHelper.tryCatch(() =>
+          signer.sendTransaction({
+            ...transactionParams,
+            chainId,
+            nonce,
+            gasLimit,
+            maxPriorityFeePerGas: gasPrice,
+            maxFeePerGas: gasPrice,
+          })
+        )
```

Comment:
What do you think about validating if the error is a cached transaction error? If yes, continue the flow, if no, throw the error.